### ISSUE TITLE
Public/Private view

### ISSUE THREAD

**Issue by jangrewe** [TwiN/gatus#638]
### Describe the feature request

Hi, after discovering that Uptime Kuma doesn't support config files (WTF?), i luckily came across Gatus and knew immediately that this is what i actually want to run! <3 

The one feature i'd really like to see would be being able to keep the status page publicly accessible, while also enabling authentication and providing an extended list of checks that are not publicly visible - e.g. anonymous users see only website endpoints, authenticated users see services, private websites, DNS, etc.
This could be done by adding another field to the config dict for an endpoint, e.g. `public: False`.

### Why do you personally want this feature to be implemented?

I'm (obviously) self-hosting a metric shit ton of stuff and want to monitor everything, but i also don't want everyone to see everything i host. I could just run 2 instances of Gatus with separate config files, but that's not the "nice" solution. ;-)

### How long have you been using this project?

1 hour

**Comment by TwiN** [comment]
Good idea - enabling authentication via `security.basic` or `security.oidc` could be tweaked so that:
- If all endpoints are private, use the current login behavior
- If some endpoints are private, but not all, show a login button somewhere on the dashboard
- If no endpoints are marked as private, use the current login behavior (for backward compatibility)

**Comment by jangrewe** [comment]
Yes, you summarized it exactly how i would love to see it! 😃 
By now i'v been using Gatus for >2 weeks already, and there are some other minor feature requests mentioned in the "introduction" discussion (especially a toggle "show only unhealthy/healthy/all), but this feature to make a Gatus instance semi-private is my biggest wishlist item. 😍

**Comment by ivan-zaitsev** [comment]
@TwiN It would be good to have a feature to restrict specific group based on security. For example in OIDC there is a claim called "groups" which can be used for this feature. For basic security I suppose it can be resolved by creating multiple credentials which would have different groups assigned. Currently it is only possible to achieve this by deploying different instance of application per group.

**Comment by xeruf** [comment]
I would love that as well, also a recent convert from Kuma :)

**Comment by tigattack** [comment]
This would be super useful. I also switched over from Uptime Kuma a couple of days ago, and I'm loving Gatus so far, but this would be a really nice addition!

**Comment by SkilledAlpaca** [comment]
One thing that Uptime Kuma did that I greatly miss, is the dashboards by url. 

I could host `internal.domain.com` and `external.domain.com` each with their own endpoints.

**Comment by TwiN** [comment]
Totally open to this feature if somebody wants to give it a try.

I think having a per-endpoint configuration that allows marking specific endpoints as private/requiring authentication would be the way to go.

**Comment by loispostula** [comment]
in #1328 I've started giving a shot to this feature.

My idea was:

1. Add a parameter on `endpoint`, `external_endpoint` and `suites` to mark them as Public.
2. Tweak the security config to allow authentication to be set either at `global` level, or `endpoint` level (Should probably be called `resource` Level)
3. If the auth is at `endpoint` level, then the auth middlewares doesn't returns 401 but the response from the protected API are filtered.
4. Finally the UI should be adapted to have a login form to display the base ui if not authenticated

**Comment by CherryDT** [comment]
I also think it would be incredibly useful to have individual dashboards on different URLs and/or with different credentials that would have different settings in regards to what a user can see. (Not just one private/public flag.) This would allow us to have a full overview over monitoring but show dashboards for individual products publicly and/or give certain clients custom access to the metrics related to their products.

**Comment by nickwebcouk** [comment]
Just want to check the status of this - I see the linked pul - #1328 the last commit was 6 months ago. I was initially just looking for a way to have a url for a specific group then I would control who accesses it through my own ACL system - but this also works!

**Comment by loispostula** [comment]
I can find some time this week to revive the pr, happy to bring needed feature in it if it make sense
